### DISCUSSION TITLE
Would like to use for a tool

### DISCUSSION THREAD

**Discussion by ghost** [xoreos/xoreos#566] [General]
I would like to use this for rendering to make tools for Neverwinter Nights. I do not know if I should just try and see if I can figure out how to do it. I guess, I want to have permission to do it but I do not want to interfere with your team or anything. If somehow, I manage to do something useful you can possibly integrate the changes back in a much better way later.

**Comment by DrMcCoy** [comment]
As long as you adhere to the GPLv3 license, feel free to do anything you like. :)

If you want to write something to integrate back into the xoreos project (and we would love to have more tools here, yes), there's a few caveats:
- In general, it makes sense to write it in C++, so that it integrates neatly with the existing codebase (there are plans to create a library with C bindings that could be used by projects in other languages, but this is probably far into the future still)
- It should be portable, at least Linux, Windows, macOS, but hopefully more platforms. I.e. keep with portable libraries and don't use platform-exclusive code without an abstraction that allows for extension for other platforms
- Keeping with the general guidelines, including code and commit style, code organization, open culture ideals, etc. would be very helpful. See https://wiki.xoreos.org/index.php?title=Developer_Central and the pages that links for a bit of an overview

Feel free to join the IRC channel if you have any questions, want to discuss ideas, etc. I can't guarantee I'll answer right away, since RL is a bit busy, but I'll try.

**Comment by DrMcCoy** [comment]
(I went ahead and converted this issue into a discussion, since this is essentially a question and not an issue.)

Sad to see you couldn't get into the xoreos codebase. Please feel free to ask for help if you still would like to try again in the future. In either case, good luck with your future endeavours! :)
